### PR TITLE
Fix #14 by adding tests

### LINKED ISSUE #14: Tests tests tests
Once we have a cross-plat xunit runner on `dotnet`, we need tests:

1. Port https://github.com/aspnet/dnx/blob/652d23d6bac5403ae7999a3cad51e25caa7eac87/test/Microsoft.Dnx.Host.Tests/RuntimeEnvironmentTests.cs
2. Test `/etc/os-release` parsing
3. Test Darwin and NT Version conversions.

### PR DESCRIPTION
**Pull request by anurse**
This does also add one small breaking change that I will publish to Announcements. Previously, when running on a Linux Distro that has a space in it's name (i.e. "Elementary OS"), we would produce an invalid RID: `elementary os.1.0-x64`. This change causes spaces to be removed, so the RID would be `elementaryos.1.0-x64`

/cc @pakrym @davidfowl @pranavkm 

Fixes #14

### CODE REVIEW

**Comment by davidfowl** [comment]
Why not use RuntimeInformation for all of the other things

**Comment by pranavkm** [comment]
I thought the plan was to delete most of this code and replace it with `RuntimeInformation`? @Eilon was ok with us removing RuntimeInfoMiddleware which was essentially the only piece of code that used most of these features.

File: src/Microsoft.Extensions.PlatformAbstractions/Internal/PlatformApis.cs [hunk 1]
```diff
@@ -2,19 +2,24 @@
...
         private static readonly Lazy<DistroInfo> _distroInfo = new Lazy<DistroInfo>(LoadDistroInfo);
 
+        public static string GetArch()
+        {
+#if NET451
+            return Environment.Is64BitProcess ? "x64" : "x86";
+#elif NETSTANDARD1_3
+            return RuntimeInformation.ProcessArchitecture.ToString().ToLowerInvariant();
+#else
+            return IntPtr.Size == 8 ? "x64" : "x86";
```
Comment (moozzyk) [inline src/Microsoft.Extensions.PlatformAbstractions/Internal/PlatformApis.cs]: ARM?

**Comment by anurse** [comment]
> Why not use RuntimeInformation for all of the other things

Not supported on .NET 4.5.1 ? Do we care?

File: src/Microsoft.Extensions.PlatformAbstractions/Internal/PlatformApis.cs (hunk 1 above)
Comment (anurse) [inline src/Microsoft.Extensions.PlatformAbstractions/Internal/PlatformApis.cs]: It was already a problem with the code before. I looked in to P/Invokes to solve it but didn't think it was worth the effort right now.

**Comment by pranavkm** [comment]
It should be fixed for net451 in rc3.

**Comment by anurse** [comment]
> It should be fixed for net451 in rc3.

That would probably solve the issues, as long as it runs on Mono. We'll still need somewhere for the RID generation to live.

**Comment by anurse** [comment]
How about ApplicationEnvironment? We moved most of that to HostingEnvironment right?

**Comment by pranavkm** [comment]
Some projects use it it to locate the `ApplicationBasePath`. We could potentially move it to common and drop this repo entirely.

**Comment by pranavkm** [comment]
The rid generation stuff was copied over to CLI - https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.InternalAbstractions/RuntimeEnvironment.cs. I changed our tools to switch to this type so it should be safe to drop the code in here.

**Pull request closed without merging**